openHarmony/classes/oNode.js

```diff
@@ -99,6 +99,19 @@
  * var attributes = myNode.attributes;
  */
 function oNode ( path, oSceneObject ){
+  // When called directly, delegate to per-type subclass so shorthand getters live on the prototype
+  if (this.constructor === oNode) {
+    var _type = node.type(path);
+    var TypeClass = oNodeTypes.getClassForType(_type);
+    if (TypeClass !== oNode) {
+      return new TypeClass(path, oSceneObject);
+    }
+  }
+  return oNode._init.call(this, path, oSceneObject);
+}
+
+// Base initializer (shared by per-type subclasses)
+oNode._init = function(path, oSceneObject){
```

Comment:
Don't we have to add this on the oNode prototype instead? I think the jerry rigging of attaching $ to the object will only work if this is an instance method (otherwise you will lose access to `this.$` inside the function body)